test/kvtree_read_scatter_single_test.c

```diff
@@ -127,8 +127,8 @@ int main(int argc, char** argv)
     rc = 0;
   }
 
-end:
   kvtree_delete(&kvtree);
+end:
```

Comment:
If the error condition happens here:
```C
kvtree* kvtree = kvtree_new();
...
  if (strcmp(tmp_str, "test data") != 0) {
    printf("Got back wrong data (got %s, expected \"test data\")\n", tmp_str);
    rc = 1;
    goto end;
  }
```
Then the kvtree will not be freed.  I like @adammoody's suggestion of `kvtree* kvtree = NULL;` for the fix.